src/dsp_tools/commands/xmlupload/resource_multimedia.py

```diff
@@ -15,40 +15,53 @@
 logger = get_logger(__name__)
 
 
-def _upload_bitstream(
+def handle_media_info(
     resource: XMLResource,
+    media_previously_uploaded: bool,
     sipi_server: Sipi,
     imgdir: str,
     permissions_lookup: dict[str, Permissions],
-) -> BitstreamInfo | None:
+) -> tuple[bool, None | BitstreamInfo]:
     """
-    This function uploads a specified bitstream file to SIPI and then returns the file information from SIPI.
+    This function checks if a resource has a bitstream
 
     Args:
-        resource: resource with that has a bitstream
+        resource: resource holding the bitstream
+        media_previously_uploaded: True if the image is already in SIPI
         sipi_server: server to upload
-        imgdir: directory of the file
         permissions_lookup: dictionary that contains the permission name as string and the corresponding Python object
+        imgdir: directory of the file
 
     Returns:
-        The information from sipi which is needed to establish a link from the resource
+        If the handling of the bitstream object was successful,
+        in case of a bitstream the object or None in case of unsuccessful try.
```

Comment:
At times, the grammar of your sentences is a bit difficult to understand :-) Can you reformulate that? Or we can also do it together ;-)